pkg/iptables/metrics.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 the Kilo authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptables
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricsClientWrapper struct {
+	client              Client
+	counterWithChain    *prometheus.CounterVec
+	counterWithoutChain *prometheus.CounterVec
+}
+
+func wrapWithMetrics(client Client, protocol string, registerer prometheus.Registerer) Client {
+	counterWithChain := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "kilo_iptables_operations_total",
+		Help:        "Number of iptables operations.",
+		ConstLabels: prometheus.Labels{"protocol": protocol},
+	}, []string{
+		"operation",
+		"table",
+		"chain",
+	})
+	counterWithoutChain := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "kilo_iptables_operations_total",
+		Help:        "Number of iptables operations.",
+		ConstLabels: prometheus.Labels{"protocol": protocol},
+	}, []string{
+		"operation",
+		"table",
+	})
+	registerer.MustRegister(counterWithChain, counterWithoutChain)
```

Comment:
if we do the `registerer == nil` check here, we don't have to do it in the caller twice (for ipv4 and ipv6)

Reply:
@clive-jevons WDYT? I don't feel it's the cleanest piece of code ever to hand in a `nil` value for `registerer` here, but I have no strong feelings against it either...

Reply:
Great idea - done 😁